### PR TITLE
CCD-4955 -- Postgres v15 update to Flex DB

### PR DESCRIPTION
**Pull request by muhammad-umerji01**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/secure/RapidBoard.jspa?rapidView=1382&projectKey=CCD&view=detail&selectedIssue=CCD-4955

### Change description ###
This PR includes the necessary updates for implementing PostgreSQL version 15. Changes have been made to variables.tf, main.tf, and state.tf to accommodate the new version and its configurations.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -73,3 +73,49 @@ module "user-profile-db-v11" {
   storage_mb      = "51200"
   common_tags     = "${var.common_tags}"
 }
+module "postgresql-v15" {
```
Comment (timja) [inline infrastructure/main.tf:76]:
```suggestion
module "postgresql_v15" {
```

File: infrastructure/state.tf [hunk 2]
```diff
@@ -1,17 +1,10 @@
...
+  features {}
+  skip_provider_registration = true
+  alias                      = "postgres_network"
+  subscription_id            = var.aks_subscription_id
+}
+
 terraform {
   backend "azurerm" {}
-
-  required_providers {
```
Comment (timja) [inline infrastructure/state.tf:4]: why remove all these?

File: infrastructure/main.tf [hunk 3]
```diff
@@ -73,3 +73,49 @@ module "user-profile-db-v11" {
...
+  pgsql_version    = "15"
+  product          = var.product
+  name             = "${local.app_full_name}-postgres-db-v15"
+  pgsql_sku        = var.pgsql_sku
+  pgsql_storage_mb = var.pgsql_storage_mb
+}
+
+resource "azurerm_key_vault_secret" "POSTGRES-USER-V15" {
+  name         = "${var.component}-POSTGRES-USER-V15"
+  value        = module.postgresql-v15.username
```
Comment (timja) [inline infrastructure/main.tf:107]:
```suggestion
  value        = module.postgresql_v15.username
```

File: infrastructure/main.tf [hunk 4]
```diff
@@ -73,3 +73,49 @@ module "user-profile-db-v11" {
...
+  key_vault_id = data.azurerm_key_vault.ccd_shared_key_vault.id
+}
+
+resource "azurerm_key_vault_secret" "POSTGRES-PASS-V15" {
+  name         = "${var.component}-POSTGRES-PASS-V15"
+  value        = module.postgresql-v15.password
```
Comment (timja) [inline infrastructure/main.tf:113]:
```suggestion
  value        = module.postgresql_v15.password
```

File: infrastructure/main.tf [hunk 5]
```diff
@@ -73,3 +73,49 @@ module "user-profile-db-v11" {
...
+  key_vault_id = data.azurerm_key_vault.ccd_shared_key_vault.id
+}
+
+resource "azurerm_key_vault_secret" "POSTGRES-HOST-V15" {
+  name         = "${var.component}-POSTGRES-HOST-V15"
+  value        = module.postgresql-v15.fqdn
```
Comment (timja) [inline infrastructure/main.tf:119]:
```suggestion
  value        = module.postgresql_v15.fqdn
```

File: infrastructure/main.tf (hunk 1 above)
Comment (muhammad-umerji01) [reply infrastructure/main.tf]: Updated to module "postgresql_v15"

File: infrastructure/state.tf (hunk 2 above)
Comment (muhammad-umerji01) [reply infrastructure/state.tf:4]: Rectified in the latest commit, Thanks for picking this up.

File: infrastructure/main.tf (hunk 4 above)
Comment (muhammad-umerji01) [reply infrastructure/main.tf]: Updated accordingly, Thanks

File: infrastructure/main.tf (hunk 5 above)
Comment (muhammad-umerji01) [reply infrastructure/main.tf]: Updated accordingly, Thanks

**Review by timja — APPROVED** [review]

**Pull request merged**